pkg/controller/operators/olm/operator.go

```diff
@@ -230,91 +261,76 @@ func (a *Operator) syncDeployment(obj interface{}) (syncError error) {
 	}
 
 	return nil
 }
 
 func (a *Operator) syncRBAC(obj interface{}) (syncError error) {
-	clusterLevel := false
+	logger := log.WithField("syncing", "rbac")
+
 	switch v := obj.(type) {
 	case *rbacv1.Role:
-		log.Debugf("sync Role %s in namespace %s", v.GetName(), v.GetNamespace())
+		logger.Debugf("sync Role %s in namespace %s", v.GetName(), v.GetNamespace())
 	case *rbacv1.RoleBinding:
-		log.Debugf("sync RoleBinding %s in namespace %s", v.GetName(), v.GetNamespace())
+		logger.Debugf("sync RoleBinding %s in namespace %s", v.GetName(), v.GetNamespace())
 	case *rbacv1.ClusterRole:
-		log.Debugf("sync ClusterRole %s", v.GetName())
-		clusterLevel = true
+		logger.Debugf("sync ClusterRole %s", v.GetName())
 	case *rbacv1.ClusterRoleBinding:
-		log.Debugf("sync ClusterRoleBinding %s", v.GetName())
-		clusterLevel = true
+		logger.Debugf("sync ClusterRoleBinding %s", v.GetName())
 	default:
 		syncError = errors.New("attempted to sync non RBAC resource with RBAC sync handler")
-		log.Debugf(syncError.Error())
+		logger.Warn(syncError.Error())
 		return
 	}
 
-	if clusterLevel {
-		// Cannot requeue namespaced owner CSVs if cluster-scoped
-		return nil
-	}
-
 	// Assert as metav1.Object
 	rbac, ok := obj.(metav1.Object)
 	if !ok {
-		syncError = errors.New("casting to runtime.Object failed")
-		log.Debugf(syncError.Error())
+		syncError = errors.New("casting to metav1.Object failed")
+		logger.Warn(syncError.Error())
 		return
 	}
 
 	// Requeue all owner CSVs
-	for _, oref := range ownerutil.GetOwnersByKind(rbac, v1alpha1.ClusterServiceVersionKind) {
-		a.requeueCSV(oref.Name, rbac.GetNamespace())
+	if ownerutil.IsOwnedByKind(rbac, v1alpha1.ClusterServiceVersionKind) {
+		a.requeueOwnerCSVs(rbac)
 	}
 
 	return nil
 }
 
-func (a *Operator) syncSecrets(obj interface{}) (syncError error) {
-	secret, ok := obj.(*corev1.Secret)
-	if !ok {
-		log.Debugf("wrong type: %#v", obj)
-		return fmt.Errorf("casting Secret failed")
-	}
+func (a *Operator) syncCore(obj interface{}) (syncError error) {
+	logger := log.WithField("syncing", "core")
 
-	// Requeue all owner CSVs
-	logger := log.WithFields(log.Fields{
-		"secret":    secret.GetName(),
-		"namespace": secret.GetNamespace(),
-	})
-	for _, oref := range ownerutil.GetOwnersByKind(secret, v1alpha1.ClusterServiceVersionKind) {
-		logger.Infof("requeuing CSV %s", oref.Name)
-		// Note: If CSVs can own secrets outside of their namespace then this can result in
-		// requeuing non-existant CSVs
-		a.requeueCSV(oref.Name, secret.GetNamespace())
+	switch v := obj.(type) {
+	case *corev1.Secret:
+		logger.Debugf("syncing Secret %s in namespace %s", v.GetName(), v.GetNamespace())
+	case *corev1.Service:
+		logger.Debugf("syncing Service %s in namespace %s", v.GetName(), v.GetNamespace())
+	case *corev1.ServiceAccount:
+		logger.Debugf("syncing ServiceAccount %s in namespace %s", v.GetName(), v.GetNamespace())
+	default:
+		syncError = errors.New("attempted to sync non Core resource with Core sync handler")
```

Comment:
Should we include the `GroupVersionKind` when logging this? Also, saying "non Core" after we only checked if it's one of three different resources is disingenuous.

Reply:
That makes sense. Honestly, the separate sync functions that don't perform actions are just for the sake of logging.  As long as we can assert it's of type `metav1.Object`, we can now requeue any owner CSVs.

Reply:
Latest code looks much better!